core/src/main/scala/org/bitcoins/core/wallet/builder/RawTxFinalizer.scala

```diff
@@ -422,7 +484,77 @@ object SubtractFeeFromOutputsFinalizer {
       .dropRight(1)
       .:+((newLastOutput, lastOutputIndex))
 
-    (newOutputs ++ unchangedOutputs).sortBy(_._2).map(_._1).toVector
+    (newOutputs ++ unchangedOutputs).sortBy(_._2).map(_._1)
+  }
+}
+
+case class DualFundingInput(
+    scriptSignature: ScriptSignature,
+    maxWitnessLen: Int)
+
+/** Finalizes a dual-funded transaction given the DualFundingInputs
+  * from both parties, their change spks and the non-change spk.
+  *
+  * This includes adding the future fee of spending transactions
+  * to the non-change output as well as subtracting relevant fees
+  * from the change outputs. This finalizer filters dust outputs.
+  */
+case class DualFundingTxFinalizer(
+    offerInputs: Vector[DualFundingInput],
+    offerPayoutSPK: ScriptPubKey,
+    offerChangeSPK: ScriptPubKey,
+    acceptInputs: Vector[DualFundingInput],
+    acceptPayoutSPK: ScriptPubKey,
+    acceptChangeSPK: ScriptPubKey,
+    feeRate: SatoshisPerVirtualByte,
+    fundingSPK: ScriptPubKey)
+    extends RawTxFinalizer {
+
+  private def computeFees(
```

Comment:
A couple thoughts on this method

1. There is a lot of magic numbers (`249`,`164`, `107` etc) that would be useful to put into vals at least so there is a name associated with it 
2. Do you have any thoughts on how we can adjust `FeeRate` to allow this kind of stuff in one place rather than having to do a one off in this method? Or other utility functions if you think `FeeRate` is an inappropriate place for it.

Reply:
So the problem is that this code is specific to DLCs and not a general thing to be used in other places. Will give the numbers names though.